AdvancedDLSupport/Attributes/DelegateLifetimeAttribute.cs

```diff
@@ -0,0 +1,66 @@
+//
+//  DelegateLifetimeAttribute.cs
+//
+//  Copyright (c) 2018 Firwood Software
+//
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU Lesser General Public License for more details.
+//
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+using System;
+using JetBrains.Annotations;
+
+namespace AdvancedDLSupport
+{
+    /// <summary>
+    /// The delegate lifetime attribute.
+    /// </summary>
+    [PublicAPI, AttributeUsage(AttributeTargets.Parameter | AttributeTargets.ReturnValue)]
+    public class DelegateLifetimeAttribute : Attribute
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DelegateLifetimeAttribute"/> class.
+        /// </summary>
+        /// <param name="lifetime">The delegate lifetime.</param>
+        public DelegateLifetimeAttribute(DelegateLifetime lifetime = DelegateLifetime.Persistent)
+        {
+            Lifetime = lifetime;
+        }
+
+        /// <summary>
+        /// Depicts the delegate lifetime.
+        /// </summary>
+        public enum DelegateLifetime
```

Comment:
This should probably be its own type.

Reply:
do you have an idea were the correct (directory) location for this enum would be? That's actually the reason why I took the easy way out :D

Reply:
It's all in the same namespace, but given that there's no other relevant directory I'd suggest the same folder as the attributes. Either that or you move this attribute and the enum to a new folder.

Reply:
Yes, this should absolutely be in its own file.